test/exceptions/Makefile

```diff
@@ -64,6 +64,7 @@ $(ROOT)/static_dtor.done: STDERR_EXP="dtor_called_more_than_once"
 $(ROOT)/future_message.done: STDERR_EXP="exception I have a custom message. exception exception "
 $(ROOT)/static_dtor.done: NEGATE=!
 $(ROOT)/catch_in_finally.done: STDERR_EXP="success."
+$(ROOT)/range_error.done: STDERR_EXP="index [7] exceeds array length 3"
```

Comment:
I'm surprised that this test doesn't cause a CI failure since DMD doesn't generate `ArrayIndexError` yet... Is this even run? I just took it from the original PR by @thewilsonator

Reply:
See https://github.com/dlang/druntime/pull/2377#issuecomment-441925502 and the discussion that immediately follows

Reply:
No, because it's missing from the [list of tests](https://github.com/dlang/druntime/blob/556e11aaa8f7117ddb372618e90b5250a2b5f7ec/test/exceptions/Makefile#L3-L4).

But tests for lowerings belong into DMD's test suite, the druntime tests should only verify that the hook works as expected. That easies maintenance and avoids akward workarounds like the skippable test added in `src/core/exception.d`.

Reply:
That's good advice, I removed the end-to-end test and changed the unittest to call the hooks directly.

Reply:
Thanks. The old druntime tests for `-checkaction=context` used to depend on the compiler lowerings and that was quite annoying to work with.